pkg/security/security.go

```diff
@@ -351,12 +352,41 @@ type Caller struct {
 	Identities []string
 }
 
+// AuthResult is the result of authentication action.
+type AuthResult struct {
+	Identities []string
+	Messages   []string
+}
+
+// AuthContext carries the contextual information and is passed across authenticators.
+type AuthContext struct {
+	// RequestContext is the context from request.
+	RequestContext context.Context
+	// Messages contains list of messages that authenticator wants to record.
+	Messages map[string][]string
+	// Caller indicates the caller that made authentication request.
+	Caller *Caller
```

Comment:
These are redundant as Authenticate returns the caller or error

Reply:
We plan to remove this next PR

Reply:
Only drive by authentication context. Return values will be removed.

Reply:
Prefer distinguish input and output